coalib/bearlib/abstractions/Linter.py

```diff
@@ -316,18 +319,33 @@ def _convert_output_regex_match_to_result(self,
                 groups['origin'] = '{} ({})'.format(klass.__name__,
                                                     groups['origin'].strip())
 
-            # Construct the result.
-            return Result.from_values(
-                origin=groups.get('origin', self),
-                message=(groups.get('message', '').strip()
-                         if result_message is None else result_message),
-                file=filename,
-                severity=groups['severity'],
-                line=groups['line'],
-                column=groups['column'],
-                end_line=groups['end_line'],
-                end_column=groups['end_column'],
-                additional_info=groups.get('additional_info', '').strip())
+            # GlobalBears do not pass a filename to the function. But they can
+            # still give one through the regex
+            if filename is None:
+                filename = groups.get('filename', None)
+
+            # Construct the result. If we have a filename, we
+            # use Result.from_values otherwise generate a project
+            # scope result.
+            if filename is not None:
+                return Result.from_values(
+                    origin=groups.get('origin', self),
+                    message=(groups.get('message', '').strip()
+                             if result_message is None else result_message),
+                    file=filename,
+                    severity=groups['severity'],
+                    line=groups['line'],
+                    column=groups['column'],
+                    end_line=groups['end_line'],
+                    end_column=groups['end_column'],
+                    additional_info=groups.get('additional_info', '').strip())
+            else:
+                return Result(
+                    origin=groups.get('origin', self),
+                    message=(groups.get('message', '').strip()
+                             if result_message is None else result_message),
+                    severity=groups['severity'],
+                    additional_info=groups.get('additional_info', '').strip())
```

Comment:
yes, you can just use the normal result constructor and create the sourceranges if the filename is given, otherwise have an empty list; i.e. not use from_values at all. I don't see how that leads to duplication